### ISSUE TITLE
Creating `sf::RenderWindow` from an existing control causes an OpenGL failing on Windows.

### ISSUE THREAD

**Issue by OwlHowlinMornSky** [SFML/SFML#2457]
## Subject of the issue

On Windows, creating `sf::RenderWindow` from an existing control causes an OpenGL failing on SFML 2.6.
SFML 2.5.1 does not have this bug.

## Your environment

* Windows 11, 23403.1001.
* SFML 2.6 (lastest snapshot, windows-vc17-64, 2023-Feb-10 03:25).
* Dynamic link.
* Visual Studio 2022.

## Steps to reproduce

Tell us how to reproduce this issue. Please provide a [minimal, complete and verifiable example](https://stackoverflow.com/help/mcve), you can use the follow template as a start:

```cpp
#include <SFML/Graphics.hpp>
#include <Windows.h>

wchar_t szWindowClass[] = L"TEST_CLASS";

int main()
{
    HINSTANCE hInst = GetModuleHandleW(NULL);

    // Regist Window Class
    WNDCLASSEXW wcex = { 0 };
    wcex.cbSize = sizeof(WNDCLASSEX);
    wcex.hbrBackground = CreateSolidBrush(RGB(0x77, 0x77, 0x77));
    wcex.hCursor = LoadCursorW(nullptr, IDC_ARROW);
    wcex.hInstance = hInst;
    wcex.lpfnWndProc = myWndProc;
    wcex.lpszClassName = szWindowClass;
    wcex.style = CS_HREDRAW | CS_VREDRAW;
    RegisterClassExW(&wcex);

    // Create Window
    HWND hWnd = CreateWindowW(szWindowClass, L"TEST_TITLE", WS_OVERLAPPEDWINDOW | WS_CLIPCHILDREN | WS_CLIPSIBLINGS, CW_USEDEFAULT, CW_USEDEFAULT, 800, 600, 0, 0, hInst, 0);

    printf_s("\nTEST 1\n\n"); // Debug code

    app.create(hWnd);

    printf_s("\nTEST 2\n\n"); // Debug code

    // Message Pump
    MSG msg = { 0 };
    bool running = true;
    while (running) {
        if (PeekMessageW(&msg, 0, 0, 0, PM_REMOVE)) {
            TranslateMessage(&msg);
            DispatchMessageW(&msg);
        }
        else {
            sf::Event evt;
            while (app.pollEvent(evt)) {
                switch (evt.type) {
                case sf::Event::Closed:
                    running = false;
                    break;
                case sf::Event::Resized:
                    app.setView(sf::View({ 0, 0, (float)evt.size.width, (float)evt.size.height }));
                    break;
		default:;
                }
            }

	    app.clear(sf::Color(0xCC, 0xCC, 0xCC));
	    app.display();
        }
    }
    app.close();

    return 0;
}
```

## Expected behavior

Only debug output in console.
```

TEST 1


TEST 2

```

## Actual behavior

output:
```

TEST 1

An internal OpenGL call failed in RenderWindow.cpp(125).
Expression:
   glGetIntegerv(GLEXT_GL_FRAMEBUFFER_BINDING, reinterpret_cast<GLint*>(&m_defaultFrameBuffer))
Error description:
   GL_INVALID_OPERATION
   The specified operation is not allowed in the current state.


TEST 2

```

**Comment by eXpl0it3r** [comment]
The declaration of `app` is missing in the code example. How and where do you create the RenderWindow?

**Comment by OwlHowlinMornSky** [comment]
I'm sorry that I missed one line:
```
#include <SFML/Graphics.hpp>
#include <Windows.h>

wchar_t szWindowClass[] = L"TEST_CLASS";
sf::RenderWindow app;

int main()
{
    HINSTANCE hInst = GetModuleHandleW(NULL);

    // Regist Window Class
    WNDCLASSEXW wcex = { 0 };
    wcex.cbSize = sizeof(WNDCLASSEX);
    wcex.hbrBackground = CreateSolidBrush(RGB(0x77, 0x77, 0x77));
    wcex.hCursor = LoadCursorW(nullptr, IDC_ARROW);
    wcex.hInstance = hInst;
    wcex.lpfnWndProc = myWndProc;
    wcex.lpszClassName = szWindowClass;
    wcex.style = CS_HREDRAW | CS_VREDRAW;
    RegisterClassExW(&wcex);

    // Create Window
    HWND hWnd = CreateWindowW(szWindowClass, L"TEST_TITLE", WS_OVERLAPPEDWINDOW | WS_CLIPCHILDREN | WS_CLIPSIBLINGS, CW_USEDEFAULT, CW_USEDEFAULT, 800, 600, 0, 0, hInst, 0);

    printf_s("\nTEST 1\n\n"); // Debug code

    app.create(hWnd);

    printf_s("\nTEST 2\n\n"); // Debug code

    // Message Pump
    MSG msg = { 0 };
    bool running = true;
    while (running) {
        if (PeekMessageW(&msg, 0, 0, 0, PM_REMOVE)) {
            TranslateMessage(&msg);
            DispatchMessageW(&msg);
        }
        else {
            sf::Event evt;
            while (app.pollEvent(evt)) {
                switch (evt.type) {
                case sf::Event::Closed:
                    running = false;
                    break;
                case sf::Event::Resized:
                    app.setView(sf::View({ 0, 0, (float)evt.size.width, (float)evt.size.height }));
                    break;
		default:;
                }
            }

	    app.clear(sf::Color(0xCC, 0xCC, 0xCC));
	    app.display();
        }
    }
    app.close();

    return 0;
}
```

**Comment by eXpl0it3r** [comment]
An important one line though 😄 

You shouldn't have global SFML resources such as a window or texture etc.  
Does the error still occur once you move the app declaration into the main() function?

**Comment by OwlHowlinMornSky** [comment]
Ok, I've tried this:
```cpp
/*...*/
int main() {
	sf::Context context;
	sf::RenderWindow app;

	HINSTANCE hInst = GetModuleHandleW(NULL);
	/*...*/
}
```
and there's no error anymore.
But if no ```sf::Context context;``` there, that error still shows up.

I think it's better to add an OpenGL check in the ```create``` function.

Version 2.6 has added
```cpp
    if (priv::RenderTextureImplFBO::isAvailable())
    {
        // Retrieve the framebuffer ID we have to bind when targeting the window for rendering
        // We assume that this window's context is still active at this point
        glCheck(glGetIntegerv(GLEXT_GL_FRAMEBUFFER_BINDING, reinterpret_cast<GLint*>(&m_defaultFrameBuffer)));
    }
```
in ```create()```, which cause this problem, while version 2.5 works well without ```sf::Context context;``` before creating window in this way.

**Comment by OwlHowlinMornSky** [comment]
I seem to underestimate the importance of OpenGL contexts.
I need to restructure my code again. 😂

**Comment by eXpl0it3r** [comment]
Root cause has been found and fixed: #3469
